lua/telescope/finders/async_function_finder.lua

```diff
@@ -0,0 +1,53 @@
+local async_lib = require "plenary.async_lib"
+local async = async_lib.async
+local await = async_lib.await
+local void = async_lib.void
+
+local make_entry = require "telescope.make_entry"
+
+return function(opts)
+  local results_maker
+  local entry_maker
+  if type(opts) == 'function' then
+    results_maker = opts
+    entry_maker = make_entry.gen_from_string()
+  else
+    results_maker = opts.results_maker
+    entry_maker = opts.entry_maker or make_entry.gen_from_string()
+  end
+
+
+  local function make_results()
+    local input_results = results_maker()
+    local results = {}
+    for k, v in ipairs(input_results) do
+      local entry = entry_maker(v)
+
+      if entry then
+        entry.index = k
+        table.insert(results, entry)
+      end
+    end
+    return results
+  end
+
+  return setmetatable({
+    results = {'results'},  -- TODO
```

Comment:
Should this just be an empty table, or should maybe this key just not exist at all in this case? Not sure it it's actually accessed somewhere?